### ISSUE TITLE
Support tcsh for shell environment resolution

### ISSUE THREAD

**Issue by nmoadev** [microsoft/vscode#136034]
When opening Visual Studio Code I see the following error popup:

```
Unable to resolve your shell environment: Unexpected exit code from spawned shell (code 1, signal null)
```

* My login shell is `tcsh` and is dictated by my organization. No option to change 😐
* `tcsh` version: `tcsh 6.20.00 (Astron) 2016-11-24 (x86_64-unknown-linux) options wide,nls,dl,al,kan,sm,rh,nd,color,filec`
* Similar to the original poster the problem seems to be the way arguments are passed to the shell program
* This issue is **new** for me as of VS Code `1.61.2`
 
Here's an excerpt from  `main.log`  showing the problem:
 ```
 [2021-10-22 10:25:10.110] [main] [trace] getUnixShellEnvironment#raw 
[2021-10-22 10:25:10.110] [main] [trace] getUnixShellEnvironment#stderr Unknown option: `-lc'
Usage: tcsh [ -bcdefilmnqstvVxX ] [ argument ... ].

[2021-10-22 10:25:10.172] [main] [trace] resolveShellEnv(): running (macOS/Linux)
[2021-10-22 10:25:10.513] [main] [trace] StorageMainService: creating workspace storage (1634912710679)
[2021-10-22 10:25:10.715] [main] [trace] SharedProcess: on vscode:createSharedProcessMessageChannel
[2021-10-22 10:25:10.715] [main] [trace] IPC Object URL: Registered new channel vscode:2cae23e8-d48c-4839-97b9-055c49ad3b30.
[2021-10-22 10:25:10.759] [main] [trace] menubarService#updateMenubar 1
[2021-10-22 10:25:10.814] [main] [trace] resolveShellEnv(): running (macOS/Linux)
[2021-10-22 10:25:10.878] [main] [trace] menubarService#updateMenubar 1
[2021-10-22 10:25:10.920] [main] [trace] SharedProcess: IPC ready
[2021-10-22 10:25:10.930] [main] [trace] Main->SharedProcess#connect: connection established
[2021-10-22 10:25:10.970] [main] [trace] SharedProcess: Overall ready
[2021-10-22 10:25:11.398] [main] [trace] RequestService#request https://update.code.visualstudio.com/api/update/linux-x64/stable/6cba118ac49a1b88332f312a8f67186f7f3c1643
[2021-10-22 10:25:11.398] [main] [trace] resolveShellEnv(): running (macOS/Linux)
[2021-10-22 10:25:11.399] [main] [error] RequestService#request resolving shell environment failed Unable to resolve your shell environment: Unexpected exit code from spawned shell (code 1, signal null)
 ```

_Originally posted by @nmoadev in https://github.com/microsoft/vscode/issues/135626#issuecomment-949686733_

**Comment by Tyriar** [comment]
@joaomoreno based on the message above the right fix would be `shellArgs = ['-i', '-c', '-l];`, right now it isn't using a login shell?

**Comment by joaomoreno** [comment]
```
➜  ~ tcsh -ilc
Unknown option: `-lc'
Usage: tcsh [ -bcdefilmnqstvVxX ] [ argument ... ].
➜  ~ tcsh --help
tcsh 6.21.00 (Astron) 2019-05-08 (x86_64-unknown-linux) options wide,nls,dl,al,kan,sm,rh,nd,color,filec

[...]
-l              act as a login shell, must be the only option specified
[...]
```

From what I can tell, there is no way I could ever use `-l` for this purpose. So, we better just do `-ic` and hope for the best. It's better than the user getting an error.

**Comment by Tyriar** [comment]
@joaomoreno ah ok, FYI apparently appending hyphen to the 0th argument is the most correct/portable way to launch a login shell according to https://github.com/microsoft/vscode/issues/123332

So that would be:

```
spawn(..., {
  ...,
  argv0: `-${systemShellUnix}`
})
```

I haven't tried it yet but we'll probably end up adopting in the terminal eventually.

**Comment by joaomoreno** [comment]
Interesting, let me know how that goes.

**Comment by MLModel** [comment]
I just installed VSC for the first time (OS X Big Sur). I can't get rid of the startup complaint quoted in the OP. My shell is zsh. As a radical test I moved both .bashrc and .zshrc out of the way and created empty replacements. So there is no error in those files. What is VSC trying to do that causes this error in my null environment? Where would it be looking? Where should I look? What should I test outside of VSC? What a waste of time.
